shared/src/main/scala/mlscript/codegen/Scope.scala

```diff
@@ -146,17 +146,89 @@ class Scope(name: Str, enclosing: Opt[Scope]) {
       }
     }
 
+  /**
+   * Look up for a trait symbol locally.
+   */
+  def getTraitSymbol(name: Str): Opt[TraitSymbol] = 
+    lexicalTypeSymbols.get(name) flatMap {
+      _ match {
+        case c: TraitSymbol => S(c)
+        case _              => N
+      }
+    }
```

Comment:
```suggestion
    lexicalTypeSymbols.get(name) collect { case c: TraitSymbol => c }
```

Same for the definitions below.

Reply:
Cool idiom!